static/src/javascripts/boot.js

```diff
@@ -0,0 +1,38 @@
+(function () {
+    var guardian = window.guardian;
+
+    // Download but don't execute, just define
+    var preFetchEnhancedBundles = function () {
+        var downloaded = false;
+        var listeners = [];
+
+        if (guardian.isModernBrowser) {
+            // Order matters because define fn needs dependencies to register
+            require([
+                'js!enhanced-vendor!order',
+                'js!bootstraps/enhanced!order'
+            ], function () {
+                downloaded = true;
+                listeners.forEach(function (fn) { fn(); });
+            });
+        }
+
+        return function (callback) {
+            if (downloaded) {
+                callback();
+            } else {
+                listeners.push(callback);
+            }
+        };
+    };
+
+    var afterPreFetch = preFetchEnhancedBundles();
+
+    require(['bootstraps/standard', 'domReady!'], function () {
+        if (guardian.isModernBrowser) {
+            afterPreFetch(function () {
+                require(['bootstraps/enhanced']);
```

Comment:
does this definitely not trigger a 2nd download if the `js!` one hasn't completed?